profiles/tests/test_gql_my_profile_query.py

```diff
@@ -419,5 +466,52 @@ def test_normal_user_can_query_their_own_profiles_sensitivedata(user_gql_client)
             "sensitivedata": {"ssn": sensitive_data.ssn},
         }
     }
-    executed = user_gql_client.execute(query)
+    executed = user_gql_client.execute(query, service=service)
     assert dict(executed["data"]) == expected_data
+
+
+def test_my_profile_results_error_if_querying_fields_not_allowed(
+    user_gql_client, service
+):
+    profile = ProfileFactory(user=user_gql_client.user)
+    ServiceConnectionFactory(profile=profile, service=service)
+
+    query = """
```

Comment:
If you change this test e.g. to the following, you are able to the field permission check and get first and last name:

```
    query = """
        {
            _service {
                sdl
            }
            myProfile {
                firstName
                lastName
            }
        }
    """
```

`myProfile` is not at the 0-index but in `info.operation.selection_set.selections[1].name.value`.

Reply:
very good spot indeed. Changed the logic so it goes through all the selections.